src/main/scala/edu/ie3/osmogrid/io/input/OsmSource.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.osmogrid.io.input
+
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.scaladsl.ActorContext
+import edu.ie3.osmogrid.cfg.OsmoGridConfig
+import edu.ie3.osmogrid.cfg.OsmoGridConfig.Input.Osm
+import edu.ie3.osmogrid.exception.IllegalConfigException
+import edu.ie3.osmogrid.io.input.InputDataProvider.InputDataEvent
+import edu.ie3.osmogrid.io.input.pbf.PbfGuardian
+import edu.ie3.osmogrid.io.input.pbf.PbfGuardian.Request
+import edu.ie3.osmogrid.model.{OsmoGridModel, SourceFilter}
+
+import java.io.{File, FileInputStream}
+import java.util.UUID
+import scala.concurrent.impl.Promise
+import scala.concurrent.{ExecutionContextExecutor, Future}
+import scala.util.{Try, Using}
+
+sealed trait OsmSource {
+
+  def read(
+      filter: SourceFilter
+  ): Future[OsmoGridModel]
+
+  def close(): Unit
+
+}
+
+object OsmSource {
+
+  final case class PbfFileSource(
+      filePath: String,
+      ctx: ActorContext[_]
+  ) extends OsmSource {
+
+    def read(
+        filter: SourceFilter
+    ): Future[OsmoGridModel] = {
+
+      val pbfReader = ctx.spawn(
+        PbfGuardian.apply(new File(filePath), filter),
+        name = s"pbf-reader-guardian-${UUID.randomUUID()}"
+      )
+
+      import akka.actor.typed.scaladsl.AskPattern._
+      import akka.util.Timeout
+      import concurrent.duration.DurationInt
+
+      // 3 hours should be more than sufficient - even for very large files on small computers
+      implicit val timeout: Timeout = 10800.seconds
+      implicit val system: ActorSystem[_] = ctx.system
+      implicit val ec: ExecutionContextExecutor = system.executionContext
+
+      val result: Future[PbfGuardian.Response] =
+        pbfReader.ask(sender => PbfGuardian.Run(sender))
+
+      result.flatMap {
```

Comment:
```suggestion
      pbfReader.ask(sender => PbfGuardian.Run(sender)).flatMap {
```

I think the assignment is unnecessary, right?